packages/shell-api/src/shard.ts

```diff
@@ -72,12 +79,91 @@ export default class Shard extends ShellApiClass {
       cmd.unique = unique;
     }
     const orderedCmd = options !== undefined ? { ...cmd, ...options } : cmd;
-    return this._mongo._serviceProvider.runCommandWithCheck(ADMIN_DB, orderedCmd);
+    try {
+      return await this._mongo._serviceProvider.runCommandWithCheck(ADMIN_DB, orderedCmd);
+    } catch (error) {
+      if (error.codeName === 'CommandNotFound') {
+        error.message = `${error.message}. Are you connected to mongos?`;
+      }
+      throw error;
+    }
   }
 
   @returnsPromise
   async status(verbose = false): Promise<any> {
+    await getConfigDB(this._mongo);
     const result = await getPrintableShardStatus(this._mongo, verbose);
     return new CommandResult('StatsResult', result);
   }
+
+  @returnsPromise
+  async addShard(url: string): Promise<any> {
+    assertArgsDefined(url);
+    await getConfigDB(this._mongo);
+    this._emitShardApiCall('addShard', { url });
+    return this._mongo._serviceProvider.runCommandWithCheck(ADMIN_DB, {
+      addShard: url
+    });
+  }
+
+  @returnsPromise
+  @serverVersions(['3.4.0', ServerVersions.latest])
+  async addShardToZone(shard: string, zone: string): Promise<any> {
+    assertArgsDefined(shard, zone);
+    this._emitShardApiCall('addShardToZone', { shard, zone });
+    await getConfigDB(this._mongo); // error if not connected to mongos
+    return this._mongo._serviceProvider.runCommandWithCheck(ADMIN_DB, {
+      addShardToZone: shard,
+      zone: zone
+    });
+  }
+
+  @returnsPromise
+  @serverVersions(['3.4.0', ServerVersions.latest])
+  async addShardTag(shard: string, tag: string): Promise<any> {
+    try {
+      return await this.addShardToZone(shard, tag);
+    } catch (error) {
+      if (error.codeName === 'CommandNotFound') {
+        error.message = `${error.message}. Are you connected to version > 3.4?`;
+      }
+      throw error;
+    }
+  }
+
+  @returnsPromise
+  async updateZoneKeyRange(namespace: string, min: Document, max: Document, zone: string): Promise<any> {
+    assertArgsDefined(namespace, min, max, zone);
+    this._emitShardApiCall('updateZoneKeyRange', { namespace, min, max, zone });
+
+    await getConfigDB(this._mongo);
+    return await this._mongo._serviceProvider.runCommandWithCheck(ADMIN_DB, {
+      updateZoneKeyRange: namespace,
+      min,
+      max,
+      zone
+    });
+  }
+
+  @returnsPromise
+  @serverVersions(['3.4.0', ServerVersions.latest])
+  async addTagRange(namespace: string, min: Document, max: Document, zone: string): Promise<any> {
+    assertArgsDefined(namespace, min, max, zone);
+    this._emitShardApiCall('updateZoneKeyRange', { namespace, min, max, zone });
+
+    await getConfigDB(this._mongo);
```

Comment:
If these calls are there for the side-effect of checking whether we are connected to a `mongos`, it might be worth leaving a comment about that :)

Reply:
Will do!